Projects/DbCLI/SearchDb.py

```diff
@@ -625,7 +626,7 @@ def poolFromCurs(curs, similarityMethod):
     parser.error('please either provide a query filename argument or do a data or smarts query')
 
   if len(options):
```

Comment:
This test will not work. 
```
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--foo')
_StoreAction(option_strings=['--foo'], dest='foo', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)
>>> args = parser.parse_args(['--foo', 'BAR'])
>>> aregs
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'aregs' is not defined
>>> args
Namespace(foo='BAR')
>>> len(args)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'Namespace' has no len()
>>> 
```
the Namespace object returned from `parse_args` doesn't provide the `__len__`  method.

Reply:
Used `vars(options)` to change it into a dictionary for which `len(vars(options))` is working.

Reply:
This is not the solution. You will get the number of defined arguments. Your test needs to be if  options.filename is given or not.

Reply:
Done.

Reply:
Does this work?
```
if options.filename:
      parser.error('please provide a filename argument')
```

Reply:
That does exactly the opposite.
```
if options.filename is not null:
    parser....
```